# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/mtransport/nr_socket_proxy_config.h
# Commit: 8196045d38b5
# Full Hash: 8196045d38b50e04313725215c36ba1618c50486
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2019-09-18 21:49:27
# Regressor Bug: 1569183
# File Overlap Count: 2
# Description:
#   Bug 1569183: Create an ipdl struct called WebrtcProxyConfig, to cut down on the arg count on lots of functions. r=mjf,mayhemer
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D45562
# ==============================================================================

diff -r 6e1eba5c88ad -r 8196045d38b5 media/mtransport/nr_socket_proxy_config.h
--- a/media/mtransport/nr_socket_proxy_config.h	Wed Sep 18 14:27:42 2019 +0000
+++ b/media/mtransport/nr_socket_proxy_config.h	Tue Sep 17 18:19:04 2019 +0000
@@ -8,21 +8,15 @@
 #define nr_socket_proxy_config__
 
 #include <memory>
-#include "nsString.h"
-
-class nsIPrincipal;
 
 namespace mozilla {
 namespace net {
-class LoadInfoArgs;
+class WebrtcProxyConfig;
 }
 
 class NrSocketProxyConfig {
  public:
-  typedef enum { kDisableProxy, kEnableProxy, kForceProxy } ProxyPolicy;
-
-  NrSocketProxyConfig(uint64_t aTabId, const nsCString& aAlpn,
-                      const net::LoadInfoArgs& aArgs, ProxyPolicy aProxyPolicy);
+  explicit NrSocketProxyConfig(const net::WebrtcProxyConfig& aProxyConfig);
 
   // We need to actually write the default impl ourselves, because the compiler
   // needs to know how to destroy mPrivate in case an exception is thrown, even
@@ -31,13 +25,11 @@
 
   ~NrSocketProxyConfig();
 
-  uint64_t GetTabId() const;
-  const nsCString& GetAlpn() const;
-  const net::LoadInfoArgs& GetLoadInfoArgs() const;
-  ProxyPolicy GetProxyPolicy() const;
+  const net::WebrtcProxyConfig& GetConfig() const;
+  bool GetForceProxy() const;
 
  private:
-  // LoadInfoArgs includes stuff that conflicts with nICEr includes.
+  // dom::ProxyConfig includes stuff that conflicts with nICEr includes.
   // Make it possible to include this header file without tripping over this
   // problem.
   class Private;